apps/jobs/views.py

```diff
@@ -206,37 +206,56 @@ def get_active_jobs_count(serialized_company_data):
 
         return serialized_company_data
 
-    @action(detail=True, methods=["get"])
+    @action(detail=False, methods=["post"])
     def users(self, request, pk=None):
         """
-        API Path: /api/v1/jobs/{pk}/users
+        API Path: /api/v1/jobs/users?status=job_status
```

Comment:
This path indicates that status is being passed in the query parameters. But the value is being read later as request.data update this accordingly either make this a get request with query params or change the description accordingly for a post request with status in payload. 
I would reccomend GET with query.